app/models/concerns/arclight/solr_document.rb

```diff
@@ -34,12 +28,8 @@ def parent_levels
       fetch('parent_levels_ssm', [])
     end
 
-    def parent_document
-      self.class.new fetch('parent').fetch('docs', []).first
-    end
-
     def eadid
-      fetch('ead_ssi', nil)&.strip
+      first('ead_ssi')&.strip || collection&.first('ead_ssi')&.strip
     end
```

Comment:
can you add a comment about why you fall back to looking at the collection for ead_id?